src/status_im/chat/subs.cljs

```diff
@@ -98,30 +111,96 @@
  :<- [:account/account]
  active-chats)
 
+(defn sort-message-groups
+  "Sorts message groups according to timestamp of first message in group"
+  [message-groups messages]
+  (let [message-ids (set (keys messages))]
+    (sort-by
+     (comp unchecked-negate :timestamp (partial get messages) :message-id first second)
+     (reduce (fn [acc [datemark group-messages]]
+               (let [filtered-messages (filter #(message-ids (:message-id %))
+                                               group-messages)]
+                 (if (zero? (count filtered-messages))
+                   acc
+                   (assoc acc datemark filtered-messages))))
+             {}
+             message-groups))))
+
+(defn get-last-message [messages message-groups]
+  (->> (sort-message-groups message-groups messages)
+       first
+       second
+       last
+       :message-id
+       (get messages)))
+
+(reg-sub
+ :chats/active-chats
+ :<- [:get-contacts]
+ :<- [:get-chats]
+ :<- [:contacts/blocked]
+ :<- [:account/account]
+ (fn [[contacts chats blocked-contacts {:keys [dev-mode?]}]]
```

Comment:
I was planning to come back to this whole file for a cleanup when doing the feature for mobile but since it's blocked by this flatlist bug I might include the mobile as well